common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -767,6 +770,54 @@ bool network_utils::linux_iface_is_up_and_running(const std::string &iface)
     return (false);
 }
 
+bool network_utils::linux_iface_get_speed(const std::string &iface, uint32_t &speed)
```

Comment:
Nitpick: I think it makes more sense to return the speed, instead of a success/failure boolean here, since there is apparently already a SPEED_UNKNOWN constant that's used for when the speed is indeterminate.
Effectively, you're returning the success/failure information twice.

Reply:
Also, please add doxygen comments documenting the function, its arguments and return value

Reply:
I think that returning a bool gives more information and can help distinguish the case in which speed is SPEED_UNKNOWN from that in which speed cannot be obtained. Despite this is not the case, caller might take a different decision on each case.

Reply:
Fair enough I guess